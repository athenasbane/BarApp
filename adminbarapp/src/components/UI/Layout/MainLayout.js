import React from 'react';
import { 
    Grid,
    makeStyles
 } from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import Menu from '../../Menu/Menu';
import Body from '../Body/Body';
import Footer from '../Footer/Footer';
import { connect } from 'react-redux';
import { getAuthed } from '../../../Redux/thunks/login.thunk';

const useStyles = makeStyles({
    root: {
        width: '100%'
    }
})

const MainLayout = (props) => {
    const classes = useStyles();
    const { authCheck } = props;
    const { token, authed } = props.login
    let history = useHistory();
    const localStorageToken = localStorage.getItem('token');
    React.useEffect(() => {
        if(localStorageToken) {
            authCheck(localStorageToken);
            setTimeout(() => history.push('/products'), 400);
        } else if(token && authed) {
            localStorage.setItem('token', token);
        } else {
            localStorage.removeItem('token');
        } 
    },[authCheck, history, localStorageToken, token, authed])

    return (
    <React.Fragment>
        <Grid container>
            <Grid item xs={12}>
                <Menu />
            </Grid>
        </Grid>
        <Grid container>
            <Grid item xs={12}>
                
            </Grid>
            <Grid className={classes.root} item xs={12}>
                <Body />
            </Grid>
        </Grid>
        <Grid container>
            <Grid item xs={12}>
               { props.login.authed ? <Footer /> : null}
            </Grid>
        </Grid>
    </React.Fragment>
)};

const mapStateToProps = state => ({
    login: state.login
})

const mapDispatchToProps = dispatch => ({
    authCheck: token => dispatch(getAuthed(token))
})

export default connect(mapStateToProps, mapDispatchToProps)(MainLayout);