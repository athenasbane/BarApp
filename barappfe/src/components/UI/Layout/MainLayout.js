import React from 'react';
import { 
    Grid,
 } from '@material-ui/core';

import Header from '../Header/Header'; 
import Body from '../Body/Body';
import Footer from '../Footer/Footer';
import SubMenu from '../../SubMenu/SubMenu';

const MainLayout = (props) => {

    return (
    <React.Fragment>
        <Grid container>
            <Grid item xs={12}><Header /></Grid>
        </Grid>
        <Grid container>
            <Grid item xs={12}>
                <SubMenu />
            </Grid>
            <Grid item xs={12}>
                <Body />
            </Grid>
        </Grid>
        <Grid container>
            <Grid item xs={12}>
               <Footer /> 
            </Grid>
        </Grid>
    </React.Fragment>
)};

export default MainLayout;