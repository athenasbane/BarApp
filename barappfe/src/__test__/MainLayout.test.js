import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import { MainLayout } from '../components/UI/Layout/MainLayout';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore();
const store = mockStore();

describe('MainLayout Tests', () => {
    test('renders correctly', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <MainLayout />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot();
    });
}); 