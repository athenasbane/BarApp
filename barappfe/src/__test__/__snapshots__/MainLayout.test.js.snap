// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout Tests renders correctly 1`] = `ShallowWrapper {}`;
